src/MusicTheory/ScaleClass.elm

```diff
@@ -0,0 +1,165 @@
+module MusicTheory.ScaleClass exposing
+    ( HeptatonicScaleClassIntervals
+    , HexatonicScaleClassIntervals
+    , OctatonicScaleClassIntervals
+    , PentatonicScaleClassIntervals
+    , ScaleClass(..)
+    , aeolian
+    , diminishedHalfWhole
+    , diminishedWholeHalf
+    , ionian
+    , major
+    , majorPentatonic
+    , minor
+    , minorPentatonic
+    , wholeTone
+    )
+
+import MusicTheory.Interval as Interval exposing (Interval)
+
+
+type ScaleClass
+    = PentatonicScaleClass PentatonicScaleClassIntervals
+    | HexatonicScaleClass HexatonicScaleClassIntervals
+    | HeptatonicScaleClass HeptatonicScaleClassIntervals
+    | OctatonicScaleClass OctatonicScaleClassIntervals
+
+
+type alias PentatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    }
+
+
+type alias HexatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    }
+
+
+type alias HeptatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    , seventhDegree : Interval
+    }
+
+
+type alias OctatonicScaleClassIntervals =
+    { firstDegree : Interval
+    , secondDegree : Interval
+    , thirdDegree : Interval
+    , fourthDegree : Interval
+    , fifthDegree : Interval
+    , sixthDegree : Interval
+    , seventhDegree : Interval
+    , eighthDegree : Interval
+    }
+
+
+major : ScaleClass
+major =
+    ionian
+
+
+minor : ScaleClass
+minor =
```

Comment:
Can you be more specific about what you want modes for?

The simplest way to get modes (of the major and melodic minor, for example) would be to define them here in this file. I think maybe that's the way we should start.

As far as I know, there is no such thing as a mode of a non-diatonic scale. So if we wrote a `modeOf` function that took a scale degree, we would probably have to either:
1. model diatonic scales and non-diatonic scales separately
2. return a `Result` for handling errors in the case where the `Scale` is not diatonic

Is there a reason you want a `Mode` type? Would a `Mode` work the same way as a scale?

Reply:
> Can you be more specific about what you want modes for?

Not really. I just thought it would be nice to derive them, because that reflects the domain.

But I guess if we hard code all the scales, that's fine, too.

Reply:
I agree, it would be nice!

I think if we model diatonic scales separately (which I am leaning toward now), we could write:
```
modeOf : DiatonicScale -> DiatonicScaleDegree -> DiatonicScale
```
and
```
isModeOf : DiatonicScale -> DiatonicScale -> Bool
```
Which could be nice for deriving and detecting those relationships.